File: go/api/options/zcount_options.go
// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0

package options

// The common interface for representing all the range type for Zcount command.
type ScoreRange interface {
	ToArgs() ([]string, error)
}

// This struct represents the min and max boundary for the Zcount command.
type ZCountRange struct {
	min scoreBoundary
	max scoreBoundary
}

// Create a new Zcount range.
func NewZCountRangeBuilder(min scoreBoundary, max scoreBoundary) *ZCountRange {
	return &ZCountRange{min, max}
}

func (zCountRange *ZCountRange) ToArgs() ([]string, error) {
	return []string{string(zCountRange.min), string(zCountRange.max)}, nil
}

File: csharp/sources/Valkey.Glide/Commands/Options/ZCountOptions.cs
// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0

namespace Valkey.Glide.Commands.Options;

/// <summary>
/// Represents the range options for the ZCOUNT command.
/// </summary>
/// <param name="min">The minimum score boundary.</param>
/// <param name="max">The maximum score boundary.</param>
public readonly struct ZCountRange(ScoreBoundary min, ScoreBoundary max)
{
    /// <summary>
    /// The minimum score boundary.
    /// </summary>
    public ScoreBoundary Min { get; } = min;

    /// <summary>
    /// The maximum score boundary.
    /// </summary>
    public ScoreBoundary Max { get; } = max;

    /// <summary>
    /// Converts the range to command arguments.
    /// </summary>
    /// <returns>An array of string arguments for the command.</returns>
    internal string[] ToArgs() => [Min.ToString(), Max.ToString()];
}
